### ISSUE TITLE
Issues with CMake and CUDA

### ISSUE THREAD

**Issue by science-enthusiast** [GPUPeople/spECK#3]
When running `sudo ./linuxsetup.sh`, the following error happens: No CMAKE_CUDA_COMPILER could be found. It is not getting fixed even after setting an environment variable `CUDACXX` to `/usr/local/cuda-11.2/bin/nvcc`.

**Comment by dabeschte** [comment]
Did you already set CUDA_HOME?

Take a look at this thread, sounds like the same problem: https://github.com/jetsonhacks/buildLibrealsense2TX/issues/13#issuecomment-573976359

**Comment by science-enthusiast** [comment]
Yes, I also googled about it just now. Appreciate such a quick response!

**Comment by science-enthusiast** [comment]
CUDA_HOME was already set in my case and it was set to `/usr/local/cuda`. However, I had to add the line `export PATH=$PATH:$CUDA_HOME/bin` to my .bashrc. I already had `export PATH=$PATH:/usr/local/cuda-11.2/bin` though. After this change, I started getting the error: "/usr/local/cuda/bin" is not able to compile a simple test program.

I tried many tricks to fix this error. None of them worked. One of the tricks seemed to make a little progress and gave the warning "CUDA_ARCHITECTURES is empty for target". However, the Makefile was not yet generated. Finally, I edited the CMakeLists.txt (see attachment) according to the faimGraph project. The build process worked completely.
[CMakeLists.txt](https://github.com/GPUPeople/spECK/files/6407813/CMakeLists.txt)
